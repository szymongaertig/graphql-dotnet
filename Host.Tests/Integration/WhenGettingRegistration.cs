using FluentAssertions;
using Host.Events;
using Host.Registrations;
using Host.Tests.Integration;
using Moq;
using Newtonsoft.Json.Linq;
using Xunit;

namespace Angeleo.AffiliateService.Tests.Framework;

public class WhenGettingRegistration : IClassFixture<CustomApplicationFactory>
{
    private readonly CustomApplicationFactory _factory;
    private Random _random = new Random();

    public WhenGettingRegistration(CustomApplicationFactory factory)
    {
        _factory = factory;
    }

    [Fact]
    public async Task ForNotExistingRegistration_ThenShouldReturnEmptyList()
    {
        // Arrange
        var client = _factory.CreateClient();
        var registration = Mock.Of<Registration>();

        // Act
        var result = await client.InvokeGraphQlRequestWithVariables(@"
  query getRegistrations($registrationId: Int!) {
  registrations(where: {id : {eq: $registrationId }}) {
    nodes {
      id
      creationDate
      clientId
      status
    }
  }
}
", new
        {
            registrationId = registration.Id
        });

        
        // Assert
        var nodes = (JArray)result.data["registrations"]["nodes"];
        nodes.Should().BeEmpty();
    }

    [Fact]
    public async Task ForExistingRegistration_ThenReturnsValidRegistration()
    {
        // Arrange
        var client = _factory.CreateClient();
        var registration = new Registration()
        {
            Id = _random.Next(),
            Event = new Event()
            {
                Id = _random.Next(),
                EventName = Guid.NewGuid().ToString()
            },
            ClientId = _random.Next(),
            Status = RegistrationStatus.Confirmed,
            CreationDate = DateTime.Now
        };
        
        await _factory.ArrangeRegistration(registration);

        // Act
        var result = await client.InvokeGraphQlRequestWithVariables(@"
          query getRegistrations($registrationId: Int!) {
          registrations(where: {id : {eq: $registrationId }}) {
            nodes {
              id
              creationDate
              clientId
              status
            }
          }
        }
        ", new
        {
            registrationId = registration.Id
        });

        // Assert
        var registrationId = result.data["registrations"]?["nodes"]?[0]["id"];
        var clientId = result.data["registrations"]?["nodes"]?[0]["clientId"];
        
        clientId.Value<int>().Should().Be(registration.ClientId);
        registrationId.Value<int>().Should().Be(registration.Id);
    }
}