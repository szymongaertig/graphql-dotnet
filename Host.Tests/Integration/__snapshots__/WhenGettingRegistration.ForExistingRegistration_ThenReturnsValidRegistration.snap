{
  "registrations": {
    "nodes": [
      {
        "id": 10,
        "creationDate": "2023-10-10T00:00:00+02:00",
        "clientId": 100,
        "status": "CONFIRMED",
        "event": {
          "id": 1,
          "eventName": "This is some event2"
        }
      }
    ]
  }
}
